### PR TITLE
minimesos ps command

### PR DESCRIPTION
**Pull request by frankscholten**
* Refactored commands to ClusterRepository can be mocked
* Added slaveId property to task

### CODE REVIEW

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.io.PrintStream;
+
+/**
+ * Lists tasks on the cluster
+ */
+public class CommandPs implements Command {
+
+    private ClusterRepository repository = new ClusterRepository();
+
+    private PrintStream output = System.out;
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS106)

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "ps";
+    }
+
+    @Override
+    public void execute() throws MinimesosException {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java]: MINOR Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    private ClusterRepository repository = new ClusterRepository();
+
+    private PrintStream output = System.out;
+
+    public CommandPs(PrintStream output) {
+        this.output = output;
+    }
+
+    public CommandPs() {
+        this.output = System.out;
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS106)

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosMaster.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.containersol.minimesos.cluster;
 
+import com.containersol.minimesos.state.State;
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosMaster.java]: MINOR Remove this unused import 'com.containersol.minimesos.state.State'. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMasterContainer.java [hunk 5]
```diff
@@ -79,6 +83,15 @@ protected CreateContainerCmd dockerCommand() {
     }
 
     @Override
+    public State getState() {
+        try {
+            return State.fromJSON(getStateInfoJSON().toString());
+        } catch (JsonParseException | JsonMappingException | UnirestException e) {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMasterContainer.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosAgentContainer.java [hunk 6]
```diff
@@ -111,4 +116,13 @@ protected CreateContainerCmd dockerCommand() {
         envs.put("SERVICE_IGNORE", "1");
         return envs;
     }
+
+    @Override
+    public State getState() {
+        try {
+            return State.fromJSON(getStateInfoJSON().toString());
+        } catch (JsonParseException | JsonMappingException | UnirestException e) {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosAgentContainer.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

**Comment by containersol** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 2 critical
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
+package com.containersol.minimesos.main;
+
+import com.containersol.minimesos.MinimesosException;
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java]: MINOR Remove this unused import 'com.containersol.minimesos.MinimesosException'. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+    private ClusterRepository repository = new ClusterRepository();
+
+    private PrintStream output = System.out; // NOSONAR
+
+    public CommandPs(PrintStream output) {
+        this.output = output;
+    }
+
+    public CommandPs() {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandPs.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1186)

**Comment by containersol** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by containersol** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 2 critical
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosContainer.java [hunk 9]
```diff
@@ -12,4 +13,6 @@
 
     JSONObject getStateInfoJSON() throws UnirestException;
 
+    State getState();
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosContainer.java]: Please, some JavaDoc describing the return of the method

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMasterContainer.java [hunk 10]
```diff
@@ -79,6 +83,15 @@ protected CreateContainerCmd dockerCommand() {
     }
 
     @Override
+    public State getState() {
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMasterContainer.java]: Implementation of this method in Master and Agent seem to be identical. Can it be moved to the common parent class?

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosContainer.java [hunk 11]
```diff
@@ -12,4 +13,6 @@
 
     JSONObject getStateInfoJSON() throws UnirestException;
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosContainer.java]: After you introduced `State getState()`, is `getStateInfoJSON()` used anywhere besides implementations of `getState()`. If not, `getStateInfoJSON()` can be removed from the interface

File: minimesos/src/main/java/com/containersol/minimesos/state/State.java [hunk 12]
```diff
@@ -23,6 +26,13 @@ public static State fromJSON(String jsonString) throws JsonParseException, JsonM
         }
     }
 
+    private Map<String, String> flags = new HashMap<>();
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/state/State.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1213)

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: minimesos/src/main/java/com/containersol/minimesos/state/State.java (hunk 12 above)
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/state/State.java]: Yep, class members should be listed before constructors.... isn't it annoying? ;-)

**Comment by sadovnikov** [comment]
Please, just move `private Map<String, String> flags` up and it's all good

Approved

**Pull request merged**